tests/components/devolo_home_network/test_binary_sensor.py

```diff
@@ -0,0 +1,132 @@
+    entry = configure_integration(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(f"{DOMAIN}.{FIRMWARE_UPDATE_AVAILABLE}") is not None
+    assert hass.states.get(f"{DOMAIN}.{CONNECTED_TO_ROUTER}") is None
+
+    await hass.config_entries.async_unload(entry.entry_id)
+
+
+@pytest.mark.usefixtures("mock_device", "mock_zeroconf")
+async def test_update_firmware_update_available(hass: HomeAssistant):
+    """Test state change of a firmware_update_available binary sensor device."""
+    state_key = f"{DOMAIN}.{FIRMWARE_UPDATE_AVAILABLE}"
+
+    entry = configure_integration(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(state_key)
+    assert state is not None
+    assert state.state == STATE_OFF
+
+    er = entity_registry.async_get(hass)
+    assert er.async_get(state_key).entity_category == ENTITY_CATEGORY_DIAGNOSTIC
+
+    # Emulate device failure
+    with patch(
+        "devolo_plc_api.device_api.deviceapi.DeviceApi.async_check_firmware_available",
+        side_effect=DeviceUnavailable,
+    ):
+        async_fire_time_changed(hass, dt.utcnow() + LONG_UPDATE_INTERVAL)
+        await hass.async_block_till_done()
+
+        state = hass.states.get(state_key)
+        assert state is not None
+        assert state.state == STATE_UNAVAILABLE
+
+    # Emulate state change
+    with patch(
+        "devolo_plc_api.device_api.deviceapi.DeviceApi.async_check_firmware_available",
+        new=AsyncMock(return_value={"result": "UPDATE_AVAILABLE"}),
+    ):
+        async_fire_time_changed(hass, dt.utcnow() + LONG_UPDATE_INTERVAL)
+        await hass.async_block_till_done()
+
+        state = hass.states.get(state_key)
+        assert state is not None
+        assert state.state == STATE_ON
+
+    await hass.config_entries.async_unload(entry.entry_id)
+
+
+@pytest.mark.usefixtures("mock_device", "mock_zeroconf")
+async def test_update_attached_to_router(hass: HomeAssistant):
+    """Test state change of a attached_to_router binary sensor device."""
+    state_key = f"{DOMAIN}.{CONNECTED_TO_ROUTER}"
+    entry = configure_integration(hass)
+
+    er = entity_registry.async_get(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Enable entity
+    er.async_update_entity(state_key, disabled_by=None)
+    await hass.async_block_till_done()
+    async_fire_time_changed(hass, dt.utcnow() + LONG_UPDATE_INTERVAL)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(state_key)
+    assert state is not None
+    assert state.state == STATE_OFF
+
+    assert er.async_get(state_key).entity_category == ENTITY_CATEGORY_DIAGNOSTIC
+
+    # Emulate device failure
+    with patch(
+        "devolo_plc_api.plcnet_api.plcnetapi.PlcNetApi.async_get_network_overview",
```

Comment:
```suggestion
        "homeassistant.components.devolo_home_network.devolo_plc_api.plcnet_api.plcnetapi.PlcNetApi.async_get_network_overview",
```

Usually we patch where the library is imported so we don't accidentally patch places we don't want to change

Reply:
In this case it would probably be better to adjust the `mock_device` instead

Reply:
The mock_device is just a collection of patches itself. I'm not seeing, how I could adjust it.